### PR TITLE
Add coordinating node logic

### PR DESCRIPTION
**Pull request by kaituo**
Note: since there are a lot of dependencies, I only list the main class and test code to save reviewers' time. The build will fail due to missing dependencies. I will use that PR just for review. will not merge it. Will have a big one in the end and merge once after all review PRs get approved.

*Issue #, if available:*

*Description of changes:*

This PR adds coordinating node logic. To get feature data, the coordinating node aggregates log entries into multiple entity keys and their corresponding value vectors.  Then, the coordinating node sends the key-value pair to the model hosting nodes using consistent hashing.  Finally, it collects acknowledgements or exceptions and returns them to the job scheduler.

Testing done:
1. Will add unit tests.
2. Manual testing passes.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/main/java/com/amazon/opendistroforelasticsearch/ad/transport/AnomalyResultTransportAction.java [hunk 1]
```diff
@@ -234,14 +239,106 @@ protected void doExecute(Task task, ActionRequest actionRequest, ActionListener<
...
+                                        new ActionListenerResponseHandler<>(
+                                            new EntityResultListener(node.getId(), adID),
+                                            AcknowledgedResponse::new,
+                                            ThreadPool.Names.SAME
+                                        )
+                                    );
+                            });
+                    }
+
+                    listener.onResponse(new AnomalyResultResponse(0, 0, 0, new ArrayList<FeatureData>()));
```
Comment (ylwu-amzn) [inline src/main/java/com/amazon/opendistroforelasticsearch/ad/transport/AnomalyResultTransportAction.java:318]: Always return empty anomaly result to job runner to persist ?

Comment (kaituo) [reply src/main/java/com/amazon/opendistroforelasticsearch/ad/transport/AnomalyResultTransportAction.java:318]: We need to return sth to job runner.  Is it necessary to persist it? Currently I do as you told me to have a heartbeat.

**Review by saratvemulapalli — APPROVED** [review]

File: src/main/java/com/amazon/opendistroforelasticsearch/ad/transport/AnomalyResultTransportAction.java (hunk 1 above)
Comment (ylwu-amzn) [reply src/main/java/com/amazon/opendistroforelasticsearch/ad/transport/AnomalyResultTransportAction.java:318]: Heartbeat can be persisted in anomaly detection state. 
I'm ok with this currently if no impact to frontend.

Comment (kaituo) [reply src/main/java/com/amazon/opendistroforelasticsearch/ad/transport/AnomalyResultTransportAction.java:318]: I did this for you.  If you think it is not necessary, I will remove it.

Comment (ylwu-amzn) [reply src/main/java/com/amazon/opendistroforelasticsearch/ad/transport/AnomalyResultTransportAction.java:318]: Sure, you can remove it.

**Review by ylwu-amzn — APPROVED** [review]
LGTM. Thanks for the change!

**Pull request closed without merging**